packages/api/src/WithVirtualization.tsx

```diff
@@ -0,0 +1,45 @@
+import { RestDataService } from '@syndesis/models';
+import * as React from 'react';
+import { DVFetch } from './DVFetch';
+import { IFetchState } from './Fetch';
+
+export interface IWithVirtualizationProps {
+  virtualizationId?: string;
+  initialValue?: RestDataService;
+  children(props: IFetchState<RestDataService>): any;
+}
+
+/**
+ * A component that fetches the virtualization with the specified identifier.
+ * @see [virtualizationId]{@link IWithVirtualizationProps#virtualizationId}
+ */
+export class WithVirtualization extends React.Component<
+  IWithVirtualizationProps
+> {
+  public render() {
+    return (
+      <DVFetch<RestDataService>
+        url={`workspace/dataservices/${this.props.virtualizationId}`}
```

Comment:
Same comment as before here, `virtualizationId` could be undefined. Is this right?

Reply:
That should not be optional here either. I will amend. Thanks.